code/datums/components/omen.dm

```diff
@@ -71,11 +93,23 @@
 		INVOKE_ASYNC(living_guy, TYPE_PROC_REF(/mob, emote), "scream")
 		living_guy.adjust_fire_stacks(20)
 		living_guy.ignite_mob(silent = TRUE)
-		if(!permanent)
-			qdel(src)
+		consume_omen()
 		return
 
-	if(!prob(8 * luck_mod))
+	var/effective_luck = luck_mod
+
+	// If there's nobody to witness the misfortune, make it less likely.
+	// This way, we allow for people to be able to get into hilarious situations without making the game nigh unplayable most of the time.
+
+	var/has_watchers
+	for(var/mob/M in view(world.view, get_turf(src)))
```

Comment:
this is not hot enough to warrant a single letter variable, write it out

Reply:
wait, what? single letter vars help perf????

Reply:
no they don't but some people do it like that because they think it will.

Reply:
you should still fix the single letter var